### ISSUE TITLE
Repeated Images when updating ListView/Adding new rows

### ISSUE THREAD

**Issue by cpsiaki** [facebook/react-native#8936]
Hi All,

I couldn't really find any documentation for this or any good explanation as to why or how this was happening but I figured I'd write it up since I found a pretty painless solution.

I was building a news feed that utilized a listView for each item. As new ones come in, they appear at the top. Part of the item is a little Image. As I added new rows, older ones would slide down but the images would not. Sometimes the images would stay on the same row.
Now I had a little caveat. The image was your profile picture. Some people have one, others don't, but only the ones that didn't would have their "placeholder" profile picture covered by the profile picture of the user in the row that was now below them. I'm sorry I'm not supplying screenshots, but I simply don't know how :/

Anyways, my code for deciding whether to show the default pic or a profile pic was as so:

```
<View style={styles.picContent}>
{url === '' ? <Image style={styles.pic} source={require('../images/profilepic.png')} /> : <Image style={styles.pic2} source={{uri:url}} />}
</View>
```
If 'url' has no value, display local placeholder, if it does, they have a profile picture.

Pretty? No. Best practice? No idea. But I've done similar things and they work so there it is. It simply wouldn't work for images and it would behave so bizarrely. In fact, it would correctly use the right styles on the WRONG image. Super frustrating.

I kept reading about how Listviews and datasources look for the same data and efficiently only change where they see changes in data. None of that mattered. No matter what weird changes I forced in my data to make something work, it still produced the bug.

Long story short, I changed the JSX for the two conditions. Here's the result:

```
<View style={styles.picContent}>
{url === '' ? <Image style={styles.pic} source={require('../images/profilepic.png')} ><View/></Image> : <Image style={styles.pic2} source={{uri:url}} />}
</View>
```

I simply wrapped a View with an Image tag for one picture, and I left the other as an Image by itself. I GUESS THIS TELLS LISTVIEW THAT THEY ARE INDEED DIFFERENT.

Anyways, I hope this helps somebody out there, one day. I spent entirely too much time on it but I found a super easy and perfectly adequate solution.

Cheers!

PS - Thanks everybody for making RN what it is. I love it, it's awesome, thanks so much.
